### PR TITLE
[MISC] Remove 'using namespace seqan3' from test/unit/alignment/matrix.

### LINKED ISSUE #1623: [MISC] Remove 'using namespace seqan3' from test/unit/alignment/matrix
Remove 'using namespace seqan3' from
- [x] test/unit/alignment/matrix

### PR DESCRIPTION
**Pull request by Irallia**
Resolves #1623. 
 
I could not remove
```cpp
using namespace seqan3::test::alignment::fixture;
```
, because it would have become very confusing. I moved and commented the line instead. If you have a better idea, I'll be happy to take it.

### CODE REVIEW

**Comment by smehringer** [comment]
> Resolves #1623.
> 
> I could not remove
> 
> ```c++
> using namespace seqan3::test::alignment::fixture;
> ```
> 
> , because it would have become very confusing. I moved and commented the line instead. If you have a better idea, I'll be happy to take it.

That's fine. Since `seqan3::test::alignment::fixture` is in our test namespace and not the public API we do not need to test qualified look up.

File: test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp [hunk 1]
```diff
@@ -10,9 +10,6 @@
 
 #include "../../pairwise/fixture/alignment_fixture.hpp"
 
-using namespace seqan3;
-using namespace seqan3::test::alignment::fixture;
```
Comment (simonsasse) [inline test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp]: Just a quick question:
Why do you delete "using namespace seqan3::test::alignment::fixture;" in this file and add it in the next file ?

Comment (Irallia) [reply test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp]: Yes, you're right. I will undo this change.

File: test/unit/alignment/matrix/debug_matrix_test.cpp [hunk 2]
```diff
@@ -287,25 +291,26 @@ struct debug_matrix_test : public ::testing::Test
...
-        EXPECT_EQ(matrix.at({row_index_type{8u}, column_index_type{16u}}), -8);
+        EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{8u}, seqan3::detail::column_index_type{0u}}), -8);
+        EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{8u}, seqan3::detail::column_index_type{6u}}), -4);
+        EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{8u}, seqan3::detail::column_index_type{16u}}), -8);
 
         for (size_t row = 0; row < matrix.rows(); row++)
             for (size_t col = 0; col < matrix.cols(); col++)
-                EXPECT_EQ((matrix.at({row_index_type{row}, column_index_type{col}})), scores[row * matrix.cols() + col]);
+                EXPECT_EQ((matrix.at({seqan3::detail::row_index_type{row},
+                                      seqan3::detail::column_index_type{col}})), scores[row * matrix.cols() + col]);
```
Comment (eseiler) [inline test/unit/alignment/matrix/debug_matrix_test.cpp]:
```suggestion
                EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{row}, seqan3::detail::column_index_type{col}}), 
                          scores[row * matrix.cols() + col]);
```

File: test/unit/alignment/matrix/debug_matrix_test.cpp [hunk 3]
```diff
@@ -314,18 +319,19 @@ struct debug_matrix_test : public ::testing::Test
...
+        EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{8u}, seqan3::detail::column_index_type{5u}}), DU);
+        EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{2u}, seqan3::detail::column_index_type{5u}}), DL);
+        EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{6u}, seqan3::detail::column_index_type{4u}}), UL);
+        EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{4u}, seqan3::detail::column_index_type{6u}}), DUL);
 
         for (size_t row = 0; row < matrix.rows(); row++)
             for (size_t col = 0; col < matrix.cols(); col++)
-                EXPECT_EQ(matrix.at({row_index_type{row}, column_index_type{col}}), traces[row * matrix.cols() + col]);
+                EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{row},
+                                     seqan3::detail::column_index_type{col}}), traces[row * matrix.cols() + col]);
```
Comment (eseiler) [inline test/unit/alignment/matrix/debug_matrix_test.cpp]:
```suggestion
                EXPECT_EQ(matrix.at({seqan3::detail::row_index_type{row}, seqan3::detail::column_index_type{col}}), 
                          traces[row * matrix.cols() + col]);
```

File: test/unit/alignment/matrix/debug_stream_alignment_coordinate_test.cpp [hunk 4]
```diff
@@ -11,27 +11,28 @@
...
-    row_incrementable co_row{detail::column_index_type{10u}, detail::row_index_type{5u}};
+    not_incrementable co_not{seqan3::detail::column_index_type{10u}, seqan3::detail::row_index_type{5u}};
+    col_incrementable co_col{seqan3::detail::column_index_type{10u}, seqan3::detail::row_index_type{5u}};
+    row_incrementable co_row{seqan3::detail::column_index_type{10u}, seqan3::detail::row_index_type{5u}};
 
-    EXPECT_TRUE((detail::is_value_specialisation_of_v<not_incrementable, detail::advanceable_alignment_coordinate>));
-    EXPECT_TRUE((detail::is_value_specialisation_of_v<col_incrementable, detail::advanceable_alignment_coordinate>));
-    EXPECT_TRUE((detail::is_value_specialisation_of_v<row_incrementable, detail::advanceable_alignment_coordinate>));
+    EXPECT_TRUE((seqan3::detail::is_value_specialisation_of_v<not_incrementable,
+                 seqan3::detail::advanceable_alignment_coordinate>));
```
Comment (eseiler) [inline test/unit/alignment/matrix/debug_stream_alignment_coordinate_test.cpp]:
```suggestion
    EXPECT_TRUE((seqan3::detail::is_value_specialisation_of_v<not_incrementable,
                                                              seqan3::detail::advanceable_alignment_coordinate>));
```

Also for the other 3 occurrences in this file

File: test/unit/alignment/matrix/detail/alignment_trace_matrix_full_banded_test.cpp [hunk 5]
```diff
@@ -36,27 +38,30 @@ struct outer_iterator
...
     {
         EXPECT_EQ(lhs.coordinate.second, std::get<0>(std::get<0>(rhs)));
         EXPECT_EQ(lhs.coordinate.first, std::get<1>(std::get<0>(rhs)));
 
-        if constexpr (std::is_same_v<std::tuple_element_t<0, test_type>,
-                                     detail::alignment_trace_matrix_full_banded<trace_directions>>)
+        using full_banded_matrix_with_trace_directions =
+            seqan3::detail::alignment_trace_matrix_full_banded<seqan3::detail::trace_directions>;
+
+        if constexpr (std::is_same_v<std::tuple_element_t<0, test_type>, full_banded_matrix_with_trace_directions>)
```
Comment (eseiler) [inline test/unit/alignment/matrix/detail/alignment_trace_matrix_full_banded_test.cpp]: If you only need the type once, I would prefer to put it into the if constexpr
```suggestion
        if constexpr (std::is_same_v<std::tuple_element_t<0, test_type>,
                                     seqan3::detail::alignment_trace_matrix_full_banded<seqan3::detail::trace_directions
                                     >>)
```

or `seqan3::detail::trace_directions>>)` in the last line and intended by 4 spaces.

File: test/unit/alignment/matrix/detail/two_dimensional_matrix_test.cpp [hunk 6]
```diff
@@ -184,45 +196,72 @@ TYPED_TEST(two_dimensional_matrix_test, subscript)
...
-    this->expect_eq(this->matrix[{row_index_type{1u}, column_index_type{0u}}], this->expected_matrix_content[4]);
-    this->expect_eq(this->matrix[{row_index_type{1u}, column_index_type{1u}}], this->expected_matrix_content[5]);
-    this->expect_eq(this->matrix[{row_index_type{1u}, column_index_type{2u}}], this->expected_matrix_content[6]);
-    this->expect_eq(this->matrix[{row_index_type{1u}, column_index_type{3u}}], this->expected_matrix_content[7]);
-    this->expect_eq(this->matrix[{row_index_type{2u}, column_index_type{0u}}], this->expected_matrix_content[8]);
-    this->expect_eq(this->matrix[{row_index_type{2u}, column_index_type{1u}}], this->expected_matrix_content[9]);
-    this->expect_eq(this->matrix[{row_index_type{2u}, column_index_type{2u}}], this->expected_matrix_content[10]);
-    this->expect_eq(this->matrix[{row_index_type{2u}, column_index_type{3u}}], this->expected_matrix_content[11]);
+    this->expect_eq(this->matrix[{seqan3::detail::row_index_type{0u},
+                                  seqan3::detail::column_index_type{0u}}], this->expected_matrix_content[0]);
```
Comment (eseiler) [inline test/unit/alignment/matrix/detail/two_dimensional_matrix_test.cpp]:
```suggestion
    this->expect_eq(this->matrix[{seqan3::detail::row_index_type{0u}, seqan3::detail::column_index_type{0u}}], 
                    this->expected_matrix_content[0]);
```
also for all the other ones (up to and including line 255)

File: test/unit/alignment/matrix/detail/alignment_trace_matrix_full_banded_test.cpp (hunk 5 above)
Comment (Irallia) [reply test/unit/alignment/matrix/detail/alignment_trace_matrix_full_banded_test.cpp]: @marehr told me to use 'using' when lines get too long and correct line breaks are unclear.

**Pull request merged**